deep_qa/training/trainer.py

```diff
@@ -684,8 +685,10 @@ def __compile_kwargs(self):
         member variables that we use to set arguments for model.compile(), we group those arguments
         together here, to only specify them once.
         """
-        return {
+        # TODO(mark): factor all compile kwargs into a single dict in the json config.
+        return Params({
+                'gradient_clipping': self.gradient_clipping,
                 'loss': self.loss,
                 'optimizer': self.optimizer,
-                'metrics': self.metrics,
```

Comment:
Keeping the last comma is just fine in python, and I think it's actually encouraged, because it makes the code easier to modify later.